clang/lib/Serialization/ASTReader.cpp

```diff
@@ -1444,6 +1444,77 @@ llvm::Error ASTReader::ReadSourceManagerBlock(ModuleFile &F) {
   }
 }
 
+std::optional<SourceLocation::UIntTy>
+ASTReader::readSLocOffset(ModuleFile *F, unsigned Index) {
```

Comment:
Does every path that returns `std::nullopt` report an error? If so I would recommend returning `Expected<...>` instead of `optional` and then report the error in the caller.